routers/web/admin/repos.go

```diff
@@ -38,6 +45,66 @@ func Repos(ctx *context.Context) {
 	})
 }
 
+func UpdateRepoPost(ctx *context.Context) {
+	temp := web.GetForm(ctx)
+	if temp == nil {
+		ctx.Data["Err_Repo_Size_Limit"] = ""
+		explore.RenderRepoSearch(ctx, &explore.RepoSearchOptions{
+			Private:          true,
+			PageSize:         setting.UI.Admin.RepoPagingNum,
+			TplName:          tplRepos,
+			OnlyShowRelevant: false,
+		})
+		return
+	}
+	form := temp.(*forms.UpdateGlobalRepoFrom)
+	ctx.Data["Title"] = ctx.Tr("admin.repositories")
+	ctx.Data["PageIsAdminRepositories"] = true
+
+	ctx.Data["RepoSizeLimit"] = form.RepoSizeLimit
+	ctx.Data["LFSSizeLimit"] = form.LFSSizeLimit
+	ctx.Data["LFSSizeInRepoSize"] = form.LFSSizeInRepoSize
+
+	repoSizeLimit, err := base.GetFileSize(form.RepoSizeLimit)
+	if err != nil {
+		ctx.Data["Err_Repo_Size_Limit"] = form.RepoSizeLimit
+		explore.RenderRepoSearch(ctx, &explore.RepoSearchOptions{
+			Private:          true,
+			PageSize:         setting.UI.Admin.RepoPagingNum,
+			TplName:          tplRepos,
+			OnlyShowRelevant: false,
+		})
+		return
+	}
+
+	lfsSizeLimit, err := base.GetFileSize(form.LFSSizeLimit)
+	if err != nil {
+		ctx.Data["Err_LFS_Size_Limit"] = form.LFSSizeLimit
+		explore.RenderRepoSearch(ctx, &explore.RepoSearchOptions{
+			Private:          true,
+			PageSize:         setting.UI.Admin.RepoPagingNum,
+			TplName:          tplRepos,
+			OnlyShowRelevant: false,
+		})
+		return
+	}
+
+	err = setting.SaveGlobalRepositorySetting(repoSizeLimit, lfsSizeLimit, form.LFSSizeInRepoSize)
```

Comment:
I don't think this will be allowed. It will cause many problems. Many users use HA mode.

Reply:
@lunny Pllease advise what is your view on this:
1) Keep how done currently, remove the save, make this a runtime only configuration
2) Modify the UI in some way (which), remove the save, make this a runtime only configuration
3) Make this a read only/display only configuration.

Reply:
At the moment, there are two possible approaches in the codebase:

- 1 Define it in the configuration file and do not allow it to be modified through the UI.
- 2 Store it in the database and make it configurable via the UI.